pkg/mosconfig/oci.go

```diff
@@ -30,6 +30,21 @@ func MountOCILayer(ocidir, name, dest string) (func(), error) {
 	}
 
 	cleanup := func() { os.RemoveAll(metadir) }
+
+	// If $ocidir/index.json exists, then it's a simple oci layout.
+	// If not, then it's a local zot layout.
+	// For image atomix/baseos:1.0.0, if it's OCI layout then the
+	// image will be called "atomix/baseos:1.0.0" in $ocidir/index.json.
+	// Otherwise, it will be called "1.0.0" in
+	// $ocidir/atomix/baseos/index.json.
+	if !PathExists(filepath.Join(ocidir, "index.json")) {
+		idx := strings.LastIndex(name, ":")
+		if idx == -1 {
+			return func() {}, fmt.Errorf("A version must be specified for zot layout")
```

Comment:
I might be ok with that, but it's a bit non-standard: if a user wants to provide a livecd with an soci rootfs that's just oci:/oci:boot , I didn't want to disable that.  Because adding a ":tag" is not only not required in ociv1, it used to be problematic (not well supported in skopeo for awhile).